File: binance4j-market/src/main/java/com/binance4j/market/dto/SymbolFilters.java
package com.binance4j.market.dto;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * The trading rules of a symbol.
 * 
 * @param icebergParts        {@link IcebergPartsFilter}.
 * @param lotSize             {@link LotSizeFilter}.
 * @param marketLotSize       {@link MarketLotSizeFilter}.
 * @param maxNumAlgoOrders    {@link MaxNumAlgoOrdersFilter}.
 * @param maxNumIcebergOrders {@link MaxNumIcebergOrdersFilter}.
 * @param maxNumOrders        {@link MaxNumOrdersFilter}.
 * @param maxPosition         {@link MaxPositionFilter}.
 * @param minNotional         {@link MinNotionalFilter}.
 * @param notional            {@link NotionalFilter}.
 * @param percentPriceBySide  {@link PercentPriceBySideFilter}.
 * @param percentPrice        {@link PercentPriceFilter}.
 * @param price               {@link PriceFilter}.
 * @param trailingDelta       {@link TrailingDeltaFilter}.
 */
@ApiModel("The trading rules of a symbol.")
public record SymbolFilters(@ApiModelProperty("IcebergParts filter") IcebergPartsFilter icebergParts, @ApiModelProperty("LotSize filter") LotSizeFilter lotSize,
		@ApiModelProperty("MarketLotSize filter") MarketLotSizeFilter marketLotSize,
		@ApiModelProperty("MaxNumAlgoOrders filter") MaxNumAlgoOrdersFilter maxNumAlgoOrders,
		@ApiModelProperty("MaxNumIcebergOrders filter") MaxNumIcebergOrdersFilter maxNumIcebergOrders,
		@ApiModelProperty("MaxNumOrders filter") MaxNumOrdersFilter maxNumOrders, @ApiModelProperty("MaxPosition filter") MaxPositionFilter maxPosition,
		@ApiModelProperty("MinNotional filter") MinNotionalFilter minNotional, @ApiModelProperty("Notional filter") NotionalFilter notional,
		@ApiModelProperty("PercentPriceBySide filter") PercentPriceBySideFilter percentPriceBySide,
		@ApiModelProperty("PercentPrice filter") PercentPriceFilter percentPrice, @ApiModelProperty("Price filter") PriceFilter price,
		@ApiModelProperty("TrailingDelta filter") TrailingDeltaFilter trailingDelta) {
}

File: src/main/kotlin/com/binance4j/connectors/market/dto/SymbolFilters.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.connectors.market.dto

import com.fasterxml.jackson.annotation.JsonProperty

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * The trading rules of a symbol.
 *
 * @property icebergParts        [IcebergPartsFilter].
 * @property lotSize             [LotSizeFilter].
 * @property marketLotSize       [MarketLotSizeFilter].
 * @property maxNumAlgoOrders    [MaxNumAlgoOrdersFilter].
 * @property maxNumIcebergOrders [MaxNumIcebergOrdersFilter].
 * @property maxNumOrders        [MaxNumOrdersFilter].
 * @property maxPosition         [MaxPositionFilter].
 * @property minNotional         [MinNotionalFilter].
 * @property notional            [NotionalFilter].
 * @property percentPriceBySide  [PercentPriceBySideFilter].
 * @property percentPrice        [PercentPriceFilter].
 * @property price               [PriceFilter].
 * @property trailingDelta       [TrailingDeltaFilter].
 */
@ApiModel("The trading rules of a symbol.")
data class SymbolFilters(
    @ApiModelProperty("IcebergParts filter")
    @JsonProperty("icebergParts")
    var icebergParts: IcebergPartsFilter,
    @ApiModelProperty("LotSize filter")
    @JsonProperty("lotSize")
    var lotSize: LotSizeFilter,
    @ApiModelProperty("MarketLotSize filter")
    @JsonProperty("marketLotSize")
    var marketLotSize: MarketLotSizeFilter,
    @ApiModelProperty("MaxNumAlgoOrders filter")
    @JsonProperty("maxNumAlgoOrders")
    var maxNumAlgoOrders: MaxNumAlgoOrdersFilter,
    @ApiModelProperty("MaxNumOrders filter")
    @JsonProperty("maxNumOrders")
    var maxNumOrders: MaxNumOrdersFilter,
    @ApiModelProperty("Min Notional filter")
    @JsonProperty("minNotional")
    var minNotional: MinNotionalFilter,
    @ApiModelProperty("PercentPrice filter")
    @JsonProperty("percentPrice")
    var percentPrice: PercentPriceFilter,
    @ApiModelProperty("Price filter")
    @JsonProperty("price")
    var price: PriceFilter,
    @ApiModelProperty("TrailingDelta filter")
    @JsonProperty("trailingDelta")
    var trailingDelta: TrailingDeltaFilter,
    @ApiModelProperty("Notional filter")
    @JsonProperty("notional")
    var notional: NotionalFilter? = null,
    @ApiModelProperty("PercentPriceBySide filter")
    @JsonProperty("percentPriceBySide")
    var percentPriceBySide: PercentPriceBySideFilter? = null,
    @ApiModelProperty("MaxNumIcebergOrders filter")
    @JsonProperty("maxNumIcebergOrders")
    var maxNumIcebergOrders: MaxNumIcebergOrdersFilter? = null,
    @ApiModelProperty("MaxPosition filter")
    @JsonProperty("maxPosition")
    var maxPosition: MaxPositionFilter? = null
)